// replit-login/dist/index.mjs
import fetch from '@replit/node-fetch';
import { parse } from 'set-cookie-parser';

const headers = {
    'User-Agent': 'Mozilla/5.0',
    'Content-Type': 'application/json',
    'X-Requested-With': 'XMLHttpRequest',
    Referrer: 'https://replit.com/',
};
async function getCookie() {
    const res = await fetch('https://replit.com/~', {
        method: 'GET',
        headers: { ...headers },
    });
    const cookies = parse(res.headers.raw()['set-cookie'], { map: true });
    return cookies['connect.sid'].value;
}
async function login(username, password, captcha) {
    const token = await getCookie();
    const res = await fetch('https://replit.com/login', {
        method: 'POST',
        headers: {
            ...headers,
            'Content-Type': 'application/json',
            Cookie: `connect.sid=${token};`,
        },
        body: JSON.stringify({
            username,
            password,
            hCaptchaResponse: captcha,
            hCaptchaSiteKey: '473079ba-e99f-4e25-a635-e9b661c7dd3e',
            teacher: false,
        }),
    }).then((res) => res.json());
    const data = {
        ...res,
        token: res.id ? token : null,
    };
    return data;
}
async function test(token) {
    const res = await fetch('https://replit.com/is_authenticated', {
        method: 'GET',
        headers: {
            ...headers,
            Cookie: `connect.sid=${token}`,
        },
    }).then((res) => res.json());
    return res;
}
async function logout(token) {
    const res = await fetch('https://replit.com/logout', {
        method: 'GET',
        headers: {
            ...headers,
            Cookie: `connect.sid=${token}`,
        },
    });
    return res.status === 200;
}

export { login, logout, test };
